### PR TITLE
Add pdm-backend

### PR DESCRIPTION
**Pull request by theAeon**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

File: recipes/pdm-backend/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  number: 0
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv
+
+requirements:
+  host:
+    - pip
+    - python >=3.7
+  run:
+    - python >=3.7
```
Comment (dopplershift) [inline recipes/pdm-backend/meta.yaml:23]: Upstream has
```
dependencies = [
    "importlib-metadata>=3.6; python_version < \"3.10\""
]
```
is this really going to work without it?

Comment (theAeon) [reply recipes/pdm-backend/meta.yaml:23]: copy-pasted from pdm-pep517 which also had that discrepancy. I assume so (but maybe worth asking the maintainer)

Comment (theAeon) [reply recipes/pdm-backend/meta.yaml:23]: @hadim

Comment (hadim) [reply recipes/pdm-backend/meta.yaml:23]: I don't recall. I suggest removing it?

Comment (theAeon) [reply recipes/pdm-backend/meta.yaml:23]: Little bit of googling and it looks like importlib-metadata was actually added to stdlib in 3.8. As conda-forge dropped support for 3.7 last year, removing it is fine.

Comment (daylinmorgan) [reply recipes/pdm-backend/meta.yaml:23]: `importlib-metadata` v1.4 was added to stdlib in 3.8. So there could be a problem if it required anything introduced in 3.10 which was `importlib-metadata` v4.6

**Review by dbast — APPROVED** [review]

**Pull request merged**